### PR TITLE
fix IntelliJ warnings

### PR DESCRIPTION
**Pull request by yilongli**
do not review yet

### CODE REVIEW

**Comment by yilongli** [comment]
Nothing major. I already restrain myself to only fixing the most obvious warnings.

@dwightguth @andreistefanescu please review

@daejunpark should I include you in pull requests regarding the java-backend? I am not entirely clear about who is in charge of what at this moment :)

**Comment by andreistefanescu** [comment]
lgtm

**Comment by daejunpark** [comment]
@yilongli You don't need to include me. :) I'm just auditing changes to keep update.

**Comment by yilongli** [comment]
@dwightguth ?

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/KILtoSMTLib.java [hunk 1]
```diff
@@ -503,7 +502,7 @@ public ASTNode transform(KItem kItem) {
             // smtlib expression instead of operator
             String expression = label;
             for (int i = 0; i < kList.getContents().size(); i++) {
-                expression = expression.replaceAll("\\#" + (i + 1) + "(?![0-9])", ((SMTLibTerm) kList.get(i).accept(this)).expression());
+                expression = expression.replaceAll("#" + (i + 1) + "(?![0-9])", ((SMTLibTerm) kList.get(i).accept(this)).expression());
```
Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/KILtoSMTLib.java:505]: you're sure # is not a special character in Java regexes?

**Comment by dwightguth** [comment]
Does this remove 100% of the unused imports in the codebase? If so, we should probably add a checkstyle for that...

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/KILtoSMTLib.java (hunk 1 above)
Comment (bmmoore) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/KILtoSMTLib.java:505]: Only under special flag settings:

http://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#COMMENTS

**Comment by dwightguth** [comment]
okay lgtm but you should probably also look into the checkstyle thing so that we don't have regressions of unused imports.

**Comment by yilongli** [comment]
@dwightguth Nope, this pull request doesn't address all the unused imports. I only run the scan over the java-backend module.

**Comment by dwightguth** [comment]
oh okay. lgtm then.

**Pull request merged**